t/param-asking.t

```diff
@@ -349,6 +349,53 @@ expect_ok "URL parsing - finish up", $cmd, [
   }
 ];
 
+
+expect_ok "IP parsing - initial prompt", $cmd, [
+  "IP Address[\r\n]{1,2}>", sub {
+    $_[0]->send("mysite.example.com\n");
+  }
+];
+expect_ok "IP parsing - used domain name", $cmd, [
+  "Specify the path", sub {
+    fail "Invalid IP address not correctly identified: used domain name";
+  }], [
+  "mysite.example.com is not a valid IPv4 address[\r\n]{1,2}>", sub {
+    $_[0]->send("123.045.067.008\n");
+  }
+];
+expect_ok "IP parsing - used leading zeros", $cmd, [
+  "Specify the path", sub {
+    fail "Invalid IP address not correctly identified: used leading zeros";
+  }], [
+  "123.045.067.008 is not a valid IPv4 address: octets cannot be zero-padded[\r\n]{1,2}>", sub {
+    $_[0]->send("123.456.789.10\n");
+  }
+];
+expect_ok "IP parsing - octets too big", $cmd, [
+  "Specify the path", sub {
+    fail "Invalid IP address not correctly identified: octets too big";
+  }], [
+  "123.456.789.10 is not a valid IPv4 address[\r\n]{1,2}>", sub {
+    $_[0]->send("123.45.67\n");
+  }
+];
+expect_ok "IP parsing - not enough octets", $cmd, [
+  "Specify the path", sub {
+    fail "Invalid IP address not correctly identified: not enough octets";
+  }], [
+  "123.45.67 is not a valid IPv4 address[\r\n]{1,2}>", sub {
+    $_[0]->send("123.4.5.6.78\n");
+  }
+];
+expect_ok "IP parsing - too many octets", $cmd, [
+  "Specify the path", sub {
+    fail "Invalid IP address not correctly identified: too many octets";
+  }], [
+  "123.4.5.6.78 is not a valid IPv4 address[\r\n]{1,2}>", sub {
+    $_[0]->send("123.4.5.67\n");
+  }
+];
+
```

Comment:
Can we add a "valid" test for something like 10.0.0.1 and 255.0.0.4?

Reply:
Yup, will do

Reply:
Ready for your re-review...

Reply:
Added bounds checks